# Question

The component below makes a mistake when using `useEffect`. When running this code, we will get an error from React! 

What is the mistake? Also, please demonstrate how to properly fetch by fixing this code.

```js
const DogDisplay = () => {
  const [imgSrc, setImgSrc] = useState('https://images.dog.ceo/breeds/hound-english/n02089973_612.jpg');

  useEffect(async () => {
    try {
      const response = await fetch('https://dog.ceo/api/breeds/image/random');
      if (!response.ok) throw new Error(`Error: ${response.status}`)
      const data = await response.json();
      setImgSrc(data.message);
    } catch (error) {
      console.error(error);
    }
  }, []);

  return <img src={imgSrc} />
}
```

# Answer
The mistake is that the code doesn't use a async function properly since `useEffect` expects a callback to be the async function.
```js
const DogDisplay = () => {
  const [imgSrc, setImgSrc] = useState('https://images.dog.ceo/breeds/hound-english/n02089973_612.jpg');

  useEffect(() => {
    Const fetchDogImage= async()=>{
      try {
      const response = await fetch('https://dog.ceo/api/breeds/image/random');
      if (!response.ok) throw new Error(`Error: ${response.status}`)
      const data = await response.json();
      setImgSrc(data.message);
    } catch (error) {
      console.error(error);
    }  
    fetchDogImage();
    }

  }, []);

  return <img src={imgSrc} />
}
```
# Teacher Feedback
3/3